test/transaction.ts

```diff
@@ -1190,15 +1208,18 @@ describe('Transaction', () => {
         transaction.begin();
 
         const expectedOptions = {readWrite: {}};
-        const {reqOpts} = stub.lastCall.args[0];
+        const {client, method, reqOpts, headers} = stub.lastCall.args[0];
 
+        assert.strictEqual(client, 'SpannerClient');
+        assert.strictEqual(method, 'beginTransaction');
         assert.deepStrictEqual(reqOpts.options, expectedOptions);
+        assert.deepStrictEqual(headers, transaction.resourceHeader_);
       });
 
       it('should accept gaxOptions', done => {
         const gaxOptions = {};
         transaction.request = config => {
-          assert.strictEqual(config.gaxOpts, gaxOptions);
+          assert.deepStrictEqual(config.gaxOpts, gaxOptions);
```

Comment:
```suggestion
          assert.strictEqual(config.gaxOpts, gaxOptions);
```

If after making that change, the test fails, then we would want to keep `deepStrictEqual`, but also set a property on `gaxOptions`.